R/module_nested_tabs.R

```diff
@@ -292,14 +291,12 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, is_module_specifi
 #' @keywords internal
 .datasets_to_data <- function(module, datasets, trigger_data = reactiveVal(1L)) {
   checkmate::assert_class(trigger_data, "reactiveVal")
-  datanames <- if (is.null(module$filter)) datasets$datanames() else module$filter
+
+  datanames <- if (is.null(module$datanames)) datasets$datanames() else module$datanames
+  datanames <- structure(datanames, names = datanames)
 
   # list of reactive filtered data
-  data <- sapply(
-    USE.NAMES = TRUE,
-    X = datanames,
-    FUN = function(x) eventReactive(trigger_data(), datasets$get_data(x, filtered = TRUE))
-  )
+  data <- lapply(datanames, function(x) eventReactive(trigger_data(), datasets$get_data(x, filtered = TRUE)))
```

Comment:
`data` needs to be a named list.
```suggestion
  data <- sapply(datanames, simplify = FALSE, function(x) eventReactive(trigger_data(), datasets$get_data(x, filtered = TRUE)))
```